lib/api/core/notifier.js

```diff
@@ -258,8 +258,42 @@ class NotifierController {
    * @param {Array} ids - list of deleted document IDs
    */
   notifyDocumentDelete(request, ids) {
-    return Bluebird.each(ids, (id, callback) => {
-      return this.kuzzle.services.list.internalCache.search(this.cacheKeyPrefix + id)
+    return Bluebird.all(ids.map(id => {
+      const
+        getRequest = new Request({
```

Comment:
@benoitvidis good point about your situation

But with the actual PR's implementation, users may be notified twice for a document's deletion, if a matched room is already in cachedRoom, or did I miss something ?

I agree that a conception workshop is needed for this point.